DevEdu.API/DevEdu.API/Startup.cs

```diff
@@ -1,156 +1,194 @@
+            services.AddScoped<IGroupRepository, GroupRepository>();
+            services.AddScoped<IPaymentRepository, PaymentRepository>();
+            services.AddScoped<IPaymentService, PaymentService>();
+            services.AddScoped<ILessonRepository, LessonRepository>();
+            services.AddScoped<ITagRepository, TagRepository>();
+            services.AddScoped<ITopicRepository, TopicRepository>();
+            services.AddScoped<IRatingRepository, RatingRepository>();
+            services.AddScoped<ICommentService, CommentService>();
+            services.AddScoped<ITagService, TagService>();
+            services.AddScoped<IGroupService, GroupService>();
+            services.AddScoped<IMaterialService, MaterialService>();
+            services.AddScoped<ITaskService, TaskService>();
+            services.AddScoped<ICourseService, CourseService>();
+            services.AddScoped<IUserService, UserService>();
+            services.AddScoped<ICourseService, CourseService>();
+            services.AddScoped<ILessonService, LessonService>();
+            services.AddScoped<ITopicService, TopicService>();
+            services.AddScoped<IRatingService, RatingService>();
+            services.AddScoped<INotificationService, NotificationService>();
+            services.AddScoped<IStudentAnswerOnTaskService, StudentAnswerOnTaskService>();
+            services.AddScoped<IAuthenticationService, AuthenticationService>();
+            services.AddScoped<IAuthOptions, AuthOptions>();
+
+            services.AddScoped<ICommentValidationHelper, CommentValidationHelper>();
+            services.AddScoped<ICourseValidationHelper, CourseValidationHelper>();
+            services.AddScoped<IGroupValidationHelper, GroupValidationHelper>();
+            services.AddScoped<ILessonValidationHelper, LessonValidationHelper>();
+            services.AddScoped<IMaterialValidationHelper, MaterialValidationHelper>();
+            services.AddScoped<INotificationValidationHelper, NotificationValidationHelper>();
+            services.AddScoped<IPaymentValidationHelper, PaymentValidationHelper>();
+            services.AddScoped<IRatingValidationHelper, RatingValidationHelper>();
+            services.AddScoped<ITagValidationHelper, TagValidationHelper>();
+            services.AddScoped<ITaskValidationHelper, TaskValidationHelper>();
+            services.AddScoped<ITopicValidationHelper, TopicValidationHelper>();
+            services.AddScoped<IUserValidationHelper, UserValidationHelper>();
+
+            services.AddControllers();
+
+            services
+                .AddMvc()
+                .AddJsonOptions(options =>
+                {
+                    options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter());
+                });
+
+
+            var provider = services.BuildServiceProvider();
+            var authOptions = provider.GetRequiredService<IAuthOptions>();

```

Comment:
I understood your problem, but building the serviceProvider inside of ConfigureServices method is not a solution
instead of this, you can create an extension method for IServiceCollection (let's name it "AddBearerAuthentication")
also, you can inject IAuthOptions in that method

in Startup you will call that method like this: services.AddBearerAuthentication()

without any kostyli